src/main/java/org/opentestsystem/shared/contentspecid/ContentSpecIdValidator.java

```diff
@@ -0,0 +1,93 @@
+package org.opentestsystem.shared.contentspecid;
+
+import org.opentestsystem.shared.contentspecid.enums.ContentSpecClaim;
+import org.opentestsystem.shared.contentspecid.enums.ContentSpecFormat;
+import org.opentestsystem.shared.contentspecid.enums.ContentSpecGrade;
+import org.opentestsystem.shared.contentspecid.enums.ContentSpecSubject;
+import org.opentestsystem.shared.contentspecid.exceptions.ErrorSeverity;
+import org.opentestsystem.shared.contentspecid.exceptions.IdValidationException;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+/**
+ * Validator for Content Spec IDs. Determines if an ID object has enough information to be formatted into
+ * the requested format.
+ */
+public class ContentSpecIdValidator {
+    public void validateFor(ContentSpecId id, ContentSpecFormat format) throws IdValidationException {
+        // Check for general validity
+        validate();
+
+        // Check for valid format and for subject match
+        boolean requiresDetail = format.isLegacy();
+
+        switch (format) {
+            case ENHANCED:
+                if (id.getSubject() == ContentSpecSubject.UNSPECIFIED) {
+                    throw new IdValidationException(ErrorSeverity.INVALID, "Subject is Unspecified.");
+                }
+                break;
+
+            case ELA_V1:
+                if (!id.isEla()) {
+                    throw new IdValidationException(ErrorSeverity.INVALID, "For format ElaV1, Subject must be ELA.");
+                }
+                break;
+
+            case MATH_V4:
+            case MATH_V5:
+            case MATH_V6:
+                if (!id.isMath()) {
+                    throw new IdValidationException(ErrorSeverity.INVALID,
+                            String.format("For format %s, Subject must be Math.", format.name()));
+                }
+                break;
+
+            default:
+                throw new IdValidationException(ErrorSeverity.INVALID, String.format("Invalid format '%s'.", format));
+        }
+
+        if (requiresDetail) {
+            if (id.getGrade() == ContentSpecGrade.UNSPECIFIED) {
+                throw new IdValidationException(ErrorSeverity.INVALID,
+                        String.format("For format %s, Grade must be specified.", format.name()));
+            }
+
+            if (id.getClaim() == ContentSpecClaim.UNSPECIFIED) {
+                throw new IdValidationException(ErrorSeverity.INVALID,
+                        String.format("For format %s, Claim must be specified.", format.name()));
+            }
+
+            if (isBlank(id.getTarget())) {
+                throw new IdValidationException(ErrorSeverity.INVALID,
+                        String.format("For format %s, Target must be specified.", format.name()));
+            }
+        }
+    }
+
+    // TODO: seems to be incorrect. There are IDs in the "good" test files that would fail this validation.
```

Comment:
Can you enhance this comment to include that this code was ported from the .NET library and a link to that github repository?  That way others will understand why this was not implemented and what the "good" ids you refer to mean.